File: src/main/java/org/tindalos/principle/domain/detector/submodulesblueprint/OverlappingSubmoduleDefinitionsException.java
package org.tindalos.principle.domain.detector.submodulesblueprint;

import java.util.Set;

import org.tindalos.principle.domain.detector.submodulesblueprint.SubmoduleDefinitions.Overlap;


@SuppressWarnings("serial")
public class OverlappingSubmoduleDefinitionsException extends InvalidBlueprintDefinitionException {

	public OverlappingSubmoduleDefinitionsException(Set<Overlap> overlaps) {
		super(toMessage(overlaps));
	}

	private static String toMessage(Set<Overlap> overlaps) {
		String msg = "Overlapping submodules: ";
		for (Overlap overlap : overlaps) {
			msg += "\n";
			for (SubmoduleId submoduleId : overlap.submoduleIds()) {
				msg += submoduleId + " and ";
			}
			msg = msg.substring(0, msg.length()-4);
		}
		return msg;
	}

}

File: src/main/java/org/tindalos/principle/domain/detector/submodulesblueprint/OverlappingSubmoduleDefinitionsException.scala
package org.tindalos.principle.domain.detector.submodulesblueprint

import scala.collection.JavaConversions.asScalaSet
import scala.collection.JavaConversions.setAsJavaSet

import org.tindalos.principle.domain.core.ListConverter

class OverlappingSubmoduleDefinitionsException(val overlaps: Set[Overlap])
  extends InvalidBlueprintDefinitionException(OverlappingSubmoduleDefinitionsException.toMessage(overlaps)) 

object OverlappingSubmoduleDefinitionsException {

  def toMessage(overlaps: java.util.Set[Overlap]) = {

    val msg = new StringBuffer("Overlapping submodules: ")
    overlaps.foreach({ overlap =>
      msg.append("\n")
      overlap.submoduleIds.foreach({ submoduleId => msg.append(submoduleId + " and ") })
      msg.append(msg.substring(0, msg.length() - 4))

    })

    msg.toString()
  }

}